sklearn/feature_selection/tests/test_mutual_info.py

```diff
@@ -158,8 +159,13 @@ def test_mutual_info_classif_mixed():
     y = ((0.5 * X[:, 0] + X[:, 2]) > 0.5).astype(int)
     X[:, 2] = X[:, 2] > 0.5
 
-    mi = mutual_info_classif(X, y, discrete_features=[2], random_state=0)
+    mi = mutual_info_classif(X, y, discrete_features=[2], n_neighbors=3,
+                             random_state=0)
     assert_array_equal(np.argsort(-mi), [2, 0, 1])
+    assert_allclose(mi, [0.06987399, 0.03197151, 0.21946924], rtol=1e-6)
+    mi_7 = mutual_info_classif(X, y, discrete_features=[2], n_neighbors=7,
+                               random_state=0)
+    assert_allclose(mi_7, [0.0735522, 0.0343685, 0.2194692], rtol=1e-5)
```

Comment:
I don't really like tests that hardcode numerical values that are true only for randomly generated but with fixed see data. Is there a better way to test the impact of `n_neighbors` without hardcoding the values?